pkg/controller/resourceclaim/controller.go

```diff
@@ -199,91 +213,170 @@ func (ec *Controller) enqueuePod(logger klog.Logger, obj interface{}, deleted bo
 	pod, ok := obj.(*v1.Pod)
 	if !ok {
 		// Not a pod?!
+		logger.Error(nil, "enqueuePod called for unexpected object", "type", fmt.Sprintf("%T", obj))
 		return
 	}
 
-	if deleted {
-		ec.deletedObjects.Add(pod.UID)
-	}
-
 	if len(pod.Spec.ResourceClaims) == 0 {
 		// Nothing to do for it at all.
 		return
 	}
 
+	if deleted {
```

Comment:
Yes. This could have been a separate PR, I just happened to notice it while working on this.

https://github.com/kubernetes/kubernetes/pull/118209/commits/f340c02a9b07b9018c48b63f945fe49c4e289ff6:
> We don't need to remember that a pod got deleted when it had no resource claims
because the code which checks the cached UIDs only checks for pods which have
resource claims.